arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Option.kt

```diff
@@ -1395,11 +1401,15 @@ public inline fun <A, B> Option<A>.redeemWith(fe: (Unit) -> Option<B>, fb: (A) -
 
 @Deprecated(
   NicheAPI + "Prefer using the Option DSL or map",
-  ReplaceWith("MA.run { this.map { List(n) { it }.fold(empty()) { acc, v -> acc + v } } }")
+  ReplaceWith("this.map { List(n) { it }.fold(empty(), combine) }")
 )
+public fun <A> Option<A>.replicate(n: Int, initial: A, combine: (A, A) -> A): Option<A> =
+  if (n <= 0) Some(initial)
+  else map { a -> List(n) { a }.fold(initial, combine) }
+
+@Deprecated(MonoidDeprecation, ReplaceWith("MA.run { replicate(n, empty()) { acc, v -> acc + v } }"))
 public fun <A> Option<A>.replicate(n: Int, MA: Monoid<A>): Option<A> = MA.run {
```

Comment:
Not sure the benefit is of adding this intermediate methods 🤔

Reply:
These were just added for consistency: since they were available with `Semigroup`, I added them with the combination functions.

Reply:
We've been deprecating `replicate` in all APIs, so I've also deprecated this one and removed the new method. I added the `NichAPI` deprecation message as well.